### PR TITLE
Convert Sonoff buttons to v2 quirk

### PR DESCRIPTION
**Pull request by TheJulianJES**
## Proposed change
This converts both existing Sonoff button quirks to a combined v2 quirk. This way we can avoid adding other signature variations.
The "signature match test" is also removed, as it doesn't make much sense with a v2 quirk(?)

## Additional information
Also addresses https://github.com/home-assistant/core/issues/104548#issuecomment-2004994990

## Checklist

- [ ] The changes are tested and work correctly
- [x] `pre-commit` checks pass / the code has been formatted using Black
- [ ] Tests have been added to verify that the new code works

### CODE REVIEW

**Comment by dmulcahey** [comment]
Why not convert these to v2 and forget the sig stuff?

**Comment by TheJulianJES** [comment]
Right, perfect candidate for that. I'll convert this tomorrow.

File: zhaquirks/sonoff/button.py [hunk 1]
```diff
@@ -1,70 +1,47 @@
...
-                    Identify.cluster_id,
-                    OnOff.cluster_id,
-                ],
+(
+    add_to_registry_v2("eWeLink", "SNZB-01P").device_automation_triggers(
+        {
+            (SHORT_PRESS, BUTTON): {
+                COMMAND: COMMAND_TOGGLE,
+                CLUSTER_ID: 6,
+                ENDPOINT_ID: 1,
```
Comment (TheJulianJES) [inline zhaquirks/sonoff/button.py]: Diff looks bad for all changes, but we can probably combine the two different Sonoff button v2 quirks, as they essentially use the same device triggers. Currently, one quirk just narrows it down to endpoint and cluster which we probably don't need.

Comment (TheJulianJES) [reply zhaquirks/sonoff/button.py]: Done in https://github.com/zigpy/zha-device-handlers/pull/3050/commits/55aa0f270c3796d38f5992113502e7dbf8b13a80

**Comment by TheJulianJES** [comment]
Codecov/project coverage decrease is expected, as a lot of code is removed.
This should be good as-is now. Merging to get it in the next quirks release for HA beta.

**Pull request merged**